examples/src/main/scala/com/cloudera/livy/examples/WordCountApp.scala

```diff
@@ -0,0 +1,179 @@
+    scalaClient = new LivyClientBuilder(true).setURI(new URI(url)).build().asScalaClient
+  }
+
+  /**
+   *  Uploads the scala-api Jar and the examples Jar from the target directory
+   *  @throws FileNotFoundException If either of scala-api Jar or examples Jar is not found
+   */
+  @throws(classOf[FileNotFoundException])
+  def uploadRelevantJarsForJobExecution(): Unit = {
+    val exampleAppJarPath = getSourcePath(this)
+    val scalaApiJarPath = getSourcePath(scalaClient)
+    uploadJar(exampleAppJarPath)
+    uploadJar(scalaApiJarPath)
+  }
+
+  @throws(classOf[FileNotFoundException])
+  private def getSourcePath(obj: Object): String = {
+    val source = obj.getClass.getProtectionDomain.getCodeSource
+    if (source != null && source.getLocation.getPath != "") {
+      source.getLocation.getPath
+    } else {
+      throw new FileNotFoundException(s"Jar containing ${obj.getClass.getName} not found.")
+    }
+  }
+
+  private def uploadJar(path: String) = {
+    val file = new File(path)
+    val uploadJarFuture = scalaClient.uploadJar(file)
+    Await.result(uploadJarFuture, 40 second) match {
+      case null => println("Successfully uploaded " + file.getName)
+    }
+  }
+
+  /**
+   * Submits a spark streaming job to the livy server
+   * <p/>
+   * The streaming job reads data from the given host and port. The data read
+   * is saved in json format as data frames in the given output path. For simplicity,
+   * the number of streaming batches are 2 with each batch for 20 seconds. The Timeout
+   * of the streaming job is set to 40 seconds
+   * @param host Hostname that Spark Streaming context has to connect for receiving data
+   * @param port Port that Spark Streaming context has to connect for receiving data
+   * @param outputPath Output path to save the processed data read by the Spark Streaming context
+   */
+  def processStreamingWordCount(host: String, port: Int,
+                                outputPath: String): ScalaJobHandle[Unit] = {
+    scalaClient.submit { context =>
+      context.createStreamingContext(20000)
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyContent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(35000)
+      ssc.stop(false, true)
+    }
+  }
+
+  /**
+   * Submits a spark sql job to the livy server
+   * <p/>
+   * The sql context job reads data frames from the given json path and executes
+   * a sql query to get the word with max count on the temp table created with data frames
+   * @param inputPath Input path to the json data containing the words
+   */
+  def getWordWithMostCount(inputPath: String): ScalaJobHandle[Row] = {
```

Comment:
I'd avoid having a job that returns a Spark-defined type such as `Row`. Even if it works, from a user's perspective, it's better to return types that are controlled by the user, to avoid serialization issues with code the user does not control.

In this case, it could just return a `String`.